lib/std/mem.zig

```diff
@@ -357,18 +357,21 @@ pub fn eql(comptime T: type, a: []const T, b: []const T) bool {
     return true;
 }
 
-pub fn len(comptime T: type, ptr: [*:0]const T) usize {
+pub fn len(comptime T: type, ptr: var) usize {
+    const sentinel: T = comptime meta.Sentinel(@TypeOf(ptr));
     var count: usize = 0;
-    while (ptr[count] != 0) : (count += 1) {}
+    while (ptr[count] != sentinel) : (count += 1) {}
     return count;
 }
 
-pub fn toSliceConst(comptime T: type, ptr: [*:0]const T) [:0]const T {
-    return ptr[0..len(T, ptr) :0];
+pub fn toSliceConst(comptime T: type, ptr: var) [:meta.Sentinel(@TypeOf(ptr))]const T {
+    const sentinel: T = comptime meta.Sentinel(@TypeOf(ptr));
+    return ptr[0..len(T, ptr) :sentinel];
 }
 
-pub fn toSlice(comptime T: type, ptr: [*:0]T) [:0]T {
-    return ptr[0..len(T, ptr) :0];
+pub fn toSlice(comptime T: type, ptr: var) [:meta.Sentinel(@TypeOf(ptr))]T {
+    const sentinel: T = comptime meta.Sentinel(@TypeOf(ptr));
+    return ptr[0..len(T, ptr) :sentinel];
 }
```

Comment:
I would like to retain the properties of these functions:
 * parameter gets implicit casted to slice (this is why element type T is provided)
 * hard-coded 0 sentinel

However I would invite you to create new functions to handle arbitrary sentinels. The less generic functions can call the more generic ones. (and format() can call the generic one.) Even with the generic ones, having the parameter implicit cast to a slice may be desirable, however I will leave that decision up to you.

But, if the parameter is `var`, no sense in having a const and non-const version, the return type should just be the correct thing. And no sense in having the element type separate either, if it can also be determined from the `var` parameter.

Reply:
Any ideas of what to call this new function?